### ISSUE TITLE
Request for Dataset Conversion Files

### ISSUE THREAD

**Issue by MadhuSaran26** [Shenzhen-Yorktal/flare23#1]
Hello,

I'm currently examining the possibility of reproducing your results. Would you be able to share the dataset conversion files for your model at your earliest convenience?

Your assistance in this matter would be greatly appreciated. Thank you in advance.

**Comment by Itsanewday** [comment]
Hi, did you mean the script for generating dataset.json? Actually, you can write a convert script using the function in utils.py in dataset_conversion. And I have upload our generate_task_json.py to dataset_conversion.

**Comment by MadhuSaran26** [comment]
Hi, thank you so much for the prompt reply. That answers my question.

I had two more quick questions. 

1. Did you use the pseudo-labels released by you in the official challenge website (aladdin5-pseudo-labels-FLARE23.zip)?
2. Are the task ids ('Task_id') in the following pre-processing steps different from each other? Can the following pre-processing steps be run in parallel?

`python nnUNet_plan_and_preprocess.py -t Task_id -pl3d ExperimentPlanner3DFabiansResUNet_Varied_Ratio -pl2d None --only_lowres`

`python nnUNet_plan_and_preprocess.py -t Task_id -pl3d ExperimentPlanner3DFabiansResUNet_v21_Tumor -pl2d None`

`python nnUNet_plan_and_preprocess.py -t Task_id -pl3d ExperimentPlanner3DFabiansResUNet_v21_Organs -pl2d None`

Your assistance in this matter would be greatly appreciated. Thank you in advance.

**Comment by Itsanewday** [comment]
Hi, we use the provided pseudo labels and you can read the [paper](https://openreview.net/pdf?id=oTI5UIgCrY) for more details. For the second problem, in my opinion, it maybe wrong. You can run them in sequence instead.

**Comment by MadhuSaran26** [comment]
Hello, I appreciate your swift response. I will review the paper again as suggested.

Moreover, I will follow your recommendation and execute the following steps sequentially. I do have one final inquiry:

Should I apply the organ and tumor preprocessing procedures to the predictions generated by the ROI model (Stage 1), or should I apply them directly to the raw data?

```python
python nnUNet_plan_and_preprocess.py -t Task_id -pl3d ExperimentPlanner3DFabiansResUNet_Varied_Ratio -pl2d None --only_lowres

python nnUNet_plan_and_preprocess.py -t Task_id -pl3d ExperimentPlanner3DFabiansResUNet_v21_Tumor -pl2d None

python nnUNet_plan_and_preprocess.py -t Task_id -pl3d ExperimentPlanner3DFabiansResUNet_v21_Organs -pl2d None
```

In addition, I am facing the following error while building the docker image as well.

_ERROR: failed to solve: failed to compute cache key: "/nnUNet/nnUNet_double_fine" not found: not found_

Could you please help me with this?

Thank you for your guidance.

**Comment by Itsanewday** [comment]
Hi，the trainging data of stage 2 were cropped from the raw data according the Bounding box of the combined labels. BTW，using the roi model to crop maybe a feasible alternative way.&nbsp;&nbsp;





---Original---
From: "Madhumitha ***@***.***&gt;
Date: Fri, Mar 1, 2024 02:30 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [Shenzhen-Yorktal/flare23] Request for Dataset Conversion Files(Issue #1)




 
Hello, I appreciate your swift response. I will review the paper again as suggested.
 
Moreover, I will follow your recommendation and execute the following steps sequentially. I do have one final inquiry:
 
Should I apply the organ and tumor preprocessing procedures to the predictions generated by the ROI model (Stage 1), or should I apply them directly to the raw data?
 python nnUNet_plan_and_preprocess.py -t Task_id -pl3d ExperimentPlanner3DFabiansResUNet_Varied_Ratio -pl2d None --only_lowres python nnUNet_plan_and_preprocess.py -t Task_id -pl3d ExperimentPlanner3DFabiansResUNet_v21_Tumor -pl2d None python nnUNet_plan_and_preprocess.py -t Task_id -pl3d ExperimentPlanner3DFabiansResUNet_v21_Organs -pl2d None
 
Thank you for your guidance.

**Comment by MadhuSaran26** [comment]
Hi, thank you so much for your reply. I really appreciate all the support that you have been providing so far.

I have one additional question to ask you.

1. The training of stage 1 is not happening because of the following issue. Could you please let me know what could be done to rectify this error?

<img width="1470" alt="Screenshot 2024-03-03 at 8 37 16 AM" src="https://github.com/Shenzhen-Yorktal/flare23/assets/117112319/c6582866-6802-470c-9390-f50acc684624">

**Comment by Itsanewday** [comment]
> Hi, thank you so much for your reply. I really appreciate all the support that you have been providing so far.
> 
> I have one additional question to ask you.
> 
> 1. The training of stage 1 is not happening because of the following issue. Could you please let me know what could be done to rectify this error?
> 
> <img alt="Screenshot 2024-03-03 at 8 37 16 AM" width="1470" src="https://private-user-images.githubusercontent.com/117112319/309572434-c6582866-6802-470c-9390-f50acc684624.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk2MDI2NzIsIm5iZiI6MTcwOTYwMjM3MiwicGF0aCI6Ii8xMTcxMTIzMTkvMzA5NTcyNDM0LWM2NTgyODY2LTY4MDItNDcwYy05MzkwLWY1MGFjYzY4NDYyNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNVQwMTMyNTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kZGYwNjk2YzJhMzQ5YzdkMzMyOTMzZGM1ZmQwOTQyYzMwYWZkMWVmYWQ1MTkzM2Q4NDI5YWUxMGIxYzg0ZTFjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.sIIEPnZZFH3hSXDpONws2sYb7JexcWE2L8YRLVAhGBo">

Hi! I'm so sorry that there a mistake in the Readme. Use `python nnunet/run/run_training.py 3d_lowres nnUNetTrainerV2_Resenc roi_task_id -f all -p nnUNetPlans_FabiansResUNet_vr` to train the ROI model.
 Since the planner of ROI model is similar to `ExperimentPlanner3DFabiansResUNet_v21` ,  the trainer should be a subclass of `nnUNetTrainerV2_ResencUNet` . 
Or you can implement a new subclass of  `ExperimentPlanner3D_v21` initialized the same as `ExperimentPlanner3DFabiansResUNet_Varied_Ratio`  and has the same `plan_experiment` and `get_target_size` of `ExperimentPlanner3DFabiansResUNet_Varied_Ratio`. In this way, you can train a generic_UNet ROI model. BTW, this is the same for stage 2. 

Best Regards.

**Comment by Itsanewday** [comment]
Hi，sorry for the late reply. It seems that you used the nnUNetTrainerV2.&nbsp; As i replied before，can you try the nnUNetTrainerV2_ResencUNet?&nbsp;
During the inference，we need the roi model to locate. So we should train a roi
&nbsp;model.

---Original---
From: "Madhumitha ***@***.***&gt;
Date: Mon, Mar 4, 2024 02:40 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [Shenzhen-Yorktal/flare23] Request for Dataset Conversion Files(Issue #1)




 
Hi, thank you so much for your reply.
 
Following up on what you said, if I don't require the output of stage 1 to train the stage 2 models, then can I skip running the stage 1 model completely?
 
The training of stage 1 is currently not happening because of the following issue. Could you please let me know what could be done here to rectify this error?
 Screenshot.2024-03-03.at.8.37.16.AM.png (view on web)

**Comment by MadhuSaran26** [comment]
Hi, thank you so much for your replies. I'll use the `nnUNetTrainerV2_ResencUNet` trainer for the ROI model, as per your suggestion.

In addition, I am facing the following error while building the docker image.

`**ERROR: failed to solve: failed to compute cache key: "/nnUNet/nnUNet_double_fine" not found: not found**`

Could you please help me understand what goes under this folder?

**Comment by Itsanewday** [comment]
Hi! Sorry i forgot the details. Maybe you should try the script in Inference. And mofidified the path in dockerfile.&nbsp;



---Original---
From: "Madhumitha ***@***.***&gt;
Date: Mon, Mar 11, 2024 12:23 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [Shenzhen-Yorktal/flare23] Request for Dataset Conversion Files(Issue #1)




 
Hi, thank you so much for your replies. I'll try your suggestions.
 
In addition, I am facing the following error while building the docker image.
 
ERROR: failed to solve: failed to compute cache key: "/nnUNet/nnUNet_double_fine" not found: not found
 
Could you please help me understand what goes under this folder?

**Comment by Itsanewday** [comment]
I checked my local code. And find that the line 12 should changed to ADD nnUNet/run_inference.py /nnUNet
and the Inference folder should be renamed to nnUNet, and you should download the pretrained models and put it to the folder models. the dockerfile should in the root Dir which has models and nnUNet(renamed by Inference). I am so sorry that if it not work. (The docker is made by my teammate.)

**Comment by MadhuSaran26** [comment]
Hi, thank you so much for all the support you provided so far.

Unfortunately, the docker file still isn't working after implementing all the suggested changes. If it's possible, could you please share the updated Dockerfile and the folder structure that should be followed?

Your assistance in this regard would greatly help me.

Meanwhile, I will continue to reproduce the results using the training suggestions that you have given. Thank you once again for your time and support.

**Comment by Itsanewday** [comment]
> Hi, thank you so much for all the support you provided so far.
> 
> Unfortunately, the docker file still isn't working after implementing all the suggested changes. If it's possible, could you please share the updated Dockerfile and the folder structure that should be followed?
> 
> Your assistance in this regard would greatly help me.
> 
> Meanwhile, I will continue to reproduce the results using the training suggestions that you have given. Thank you once again for your time and support.

Can you show more details about the docker error? And how about the training now? 
I checked the code again and update it, it should work for docker now.
